### ISSUE TITLE
Fails to extract version of "Development" Graphviz 

### ISSUE THREAD

**Issue by egberts** [xflr6/graphviz#111]
I was troubleshooting the Graphviz and using your python3-graphviz on Debian 10.4 Buster.

The version that I have is:
```console
$ dot  -V
dot - graphviz version 2.44.2~dev.20200927.0217 (20200927.0217)
```

The behavior (actual failing result) is:
```console
PyDot version: 1.4.1
Traceback (most recent call last):
  File "/home/steve/work/datascience/WSGraphDemo/bug-gv.py", line 6, in <module>
    print('Graphviz version:', graphviz.version())
  File "/home/steve/work/datascience/WSGraphDemo/venv/lib/python3.7/site-packages/graphviz/backend.py", line 265, in version
    raise RuntimeError('cannot parse %r output: %r' % (cmd, out))
RuntimeError: cannot parse ['dot', '-V'] output: 'dot - graphviz version 2.44.2~dev.20200927.0217 (20200927.0217)\n'

Process finished with exit code 1
```
The impacted line is in [`backend.py line 263`](https://github.com/xflr6/graphviz/blob/3651713c98c997ce90ba5fcc27c24837e444dcff/graphviz/backend.py#L263).

The new code that made it work has been pushed in #110

**Comment by egberts** [comment]
Versions used:

This python3-graphviz (Debian)  but deleted and using 3651713c98c997ce90ba5fcc27c24837e444dcff

Graphviz 2.44.2 (nosebleed version)
Python 3.7.3
Distro: Debian 10.4 (Buster)
OS:
```
Linux arca 4.19.0-10-amd64 #1 SMP Debian 4.19.132-1 (2020-07-24) x86_64 GNU/Linux
```

**Comment by xflr6** [comment]
> Graphviz 2.44.2 (nosebleed version)

Can you provide more information? Did you download [here](https://www.graphviz.org/pub/graphviz/development/SOURCES/webdot-working.tar.gz) or clone the repo and then built and installed it? I am wondering whether the versioning scheme of Graphviz has changed or they introduced some more markup for development versions.

```
dot - graphviz version 2.44.2~dev.20200927.0217 (20200927.0217)
```

What would be a good return value for `version()`?
- `(2, 44, 2)` (require `lax=True`?)
- `(2, 44, 2, 'dev', 20200927, 0217)`
- `(2, 44, 2, 20200927, 0217)`

**Comment by egberts** [comment]
Graphviz.version didn’t include the ‘dev’ part.

**Comment by egberts** [comment]
Mmmmm, my effort to add this version line into the `test_backend.py` and run
```console
$ pytest test_backend.py
```
resulted in unable to find `pytest.exe`.   I'm running Linux box, not Windows.
```console
~/work/github/graphviz/tests$ pytest test_backend.py 
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov --cov-report=term --cov-report=html
  inifile: /home/steve/work/github/graphviz/setup.cfg
  rootdir: /home/steve/work/github/graphviz
```

However, I see that you've already added the missing line to `test_backend.py`

**Comment by xflr6** [comment]
Try running `pytest` or `python run-tests.py` in the root dir (the one with `setup.py`).

**Comment by egberts** [comment]
It passed.

**Comment by xflr6** [comment]
Thanks.

> I cloned all repos.

Can you provide the steps you used to built/install the development version with the `dot -V` output that fails to parse (still trying to find out if there was a change in the Graphviz versioning scheme or not)?

**Comment by xflr6** [comment]
Okay, I think it's upstream [gen_version.py](https://gitlab.com/graphviz/graphviz/-/blob/master/gen_version.py) that was added recently.

**Comment by xflr6** [comment]
With the upstream definition of the version entry format, I prefer to parse it in a more strict manner (the final space in the [current regex](https://github.com/xflr6/graphviz/blob/3651713c98c997ce90ba5fcc27c24837e444dcff/graphviz/backend.py#L263) is intended).

Implementation in [this branch](https://github.com/xflr6/graphviz/compare/version) (new regex is a moutful).

I would squash and merge #110 (acknowledging your contribution) and then merge what I have above. WDYT?

**Comment by egberts** [comment]
Sure.

**Comment by xflr6** [comment]
Thanks. Fixed in [0.14.2 release](https://pypi.org/project/graphviz/0.14.2/#files) (22a61a48c2c9e30f34814dc866964b3aca2cd85e)
